src/ray/gcs/gcs_server/gcs_redis_failure_detector.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RAY_GCS_REDIS_FAILURE_DETECTOR_H
+#define RAY_GCS_REDIS_FAILURE_DETECTOR_H
+
+#include <boost/asio.hpp>
+#include "ray/gcs/redis_context.h"
+
+namespace ray {
+
+namespace gcs {
+class RedisGcsClient;
+
+/// GcsRedisFailureDetector is responsible for monitoring redis.
+class GcsRedisFailureDetector {
```

Comment:
How about abstract a base class of FailureDetector as we discussed last week ?

Reply:
Detector  implementation is relatively simple, making an abstract class may increase the complexity.